src/test/java/com/mongodb/kafka/connect/source/schema/BsonDocumentToSchemaTest.java

```diff
@@ -0,0 +1,244 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.kafka.connect.source.schema;
+
+import static com.mongodb.kafka.connect.source.schema.BsonDocumentToSchema.DEFAULT_FIELD_NAME;
+import static com.mongodb.kafka.connect.source.schema.BsonDocumentToSchema.inferDocumentSchema;
+import static com.mongodb.kafka.connect.source.schema.SchemaUtils.assertSchemaEquals;
+
+import org.apache.kafka.connect.data.Decimal;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.SchemaBuilder;
+import org.apache.kafka.connect.data.Timestamp;
+import org.junit.jupiter.api.Test;
+
+import org.bson.BsonDocument;
+
+@SuppressWarnings("ArraysAsListWithZeroOrOneArgument")
+public class BsonDocumentToSchemaTest {
```

Comment:
I ran these new tests with coverage enabled and most of the `BsonDocumentToSchema#combineFieldSchema` method is not covered by tests.  So I'm not clear why the tests that were added in the initial commit have been removed, as I think those did cover that method.

Reply:
Updated test cases to ensure full coverage of all branches